src/main/java/com/ibm/cloudant/kafka/schema/ConnectRecordMapper.java

```diff
@@ -17,20 +31,34 @@
     private static Logger LOG = LoggerFactory.getLogger(ConnectRecordMapper.class);
 
     public Map<String, Object> apply(ConnectRecord<R> record) {
+        // Check if custom header exists on the record and use the value for the document's id
+        String headerValue = getHeaderForDocId(record);
         // we can convert from a struct or a map - assume a map when a value schema is not provided
         Schema.Type schemaType = record.valueSchema() == null ? Schema.Type.MAP : record.valueSchema().type();
         Map<String, Object> toReturn = new HashMap<>();
         switch (schemaType) {
             case MAP:
                 if (record.value() instanceof Map) {
-                    return convertMap((Map) record.value(), toReturn);
+                    if (headerValue != null) {
+                        convertMap((Map) record.value(), toReturn);
+                        toReturn.put("_id", headerValue);
+                        return toReturn;
+                    } else {
+                        return convertMap((Map) record.value(), toReturn);
+                    }
```

Comment:
>Let me know if it's worth adding a !headerValue.isEmpty() check.

I think it would be worth doing there is little value in adding an empty string into the map.